src/Components/test/testassets/Components.TestServer/RazorComponents/Pages/Auth/InteractiveAuthenticationState.razor

```diff
@@ -13,8 +13,8 @@
 
 <hr />
 
-<a href="@Nav.GetUriWithQueryParameter("username", "YourUsername")">Log in</a> |
-<a href="@Nav.GetUriWithQueryParameter("username", "")">Log out</a>
+<a href="@Nav.GetUriWithQueryParameter("username", "YourUsername")" onclick="location.reload()">Log in</a> |
+<a href="@Nav.GetUriWithQueryParameter("username", "")" onclick="location.reload()">Log out</a>
```

Comment:
@SteveSandersonMS FYI, this change was required in order to make the `CanUseServerAuthenticationState_Interactive` test pass.

Before the changes in this PR, we would refresh the page instead of performing an enhanced navigation if there were interactive components on the page. Since an enhanced navigation is now performed, there isn't an opportunity for an updated `Task<AuthenticationState>` to be supplied to the `CascadingAuthenticationStateConsumer` component (because the cascading value can't cross the render mode boundary).

Reply:
Makes sense. This does seem like an actual scenario that could occur in real apps too (logic inside an SSR render might change the user's authentication state, but the circuit won't update its auth state since that's entirely separate).

I guess that:

 * For auth state changes like "log in" and "log out", you would either have to cause that to happen inside the circuit, or you'd have to forcibly reload the page like your updated test logic does here
 * For auth state changes like "user roles have changed", you would have to do the same as above if you need the change to take effect instantly, or you could wait for something like `RevalidatingServerAuthenticationStateProvider` to detect the stale state if it's OK for it to take a few minutes to update.

In any case, your change to the test logic here looks fine to me!